examples/bull-spread/bull-spread.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"cloud.google.com/go/civil"
+	"github.com/shopspring/decimal"
+
+	"github.com/alpacahq/alpaca-trade-api-go/v3/alpaca"
+	"github.com/alpacahq/alpaca-trade-api-go/v3/marketdata"
+)
+
+var decimal10k = decimal.NewFromInt(10000)
+
+const (
+	oneWeek = 7
+)
+
+type Service struct {
+	tdClient *alpaca.Client
+	mdClient *marketdata.Client
+	acct     *alpaca.Account
+}
+
+func NewService() (*Service, error) {
+	// You can set your API key/secret here or you can use environment variables
+	apiKey := os.Getenv("APCA_API_KEY_ID")
+	apiSecret := os.Getenv("APCA_API_SECRET_KEY")
+	// NOTE: mleg complex option strategies are still in beta only availabe in Paper
+	baseURL := "https://paper-api.alpaca.markets"
+
+	tdClient := alpaca.NewClient(alpaca.ClientOpts{
+		APIKey:    apiKey,
+		APISecret: apiSecret,
+		BaseURL:   baseURL,
+	})
+
+	mdClient := marketdata.NewClient(marketdata.ClientOpts{
+		APIKey:    apiKey,
+		APISecret: apiSecret,
+	})
+
+	// Cancel any open orders so they don't interfere with this algo
+	if err := tdClient.CancelAllOrders(); err != nil {
+		return nil, err
+	}
+
+	// Make sure we have enough green for some mleg fun
+	acct, err := tdClient.GetAccount()
+	if err != nil {
+		return nil, err
+	}
+	if acct.BuyingPower.LessThan(decimal10k) {
+		return nil, fmt.Errorf("insufficient buying power: needed %s, have %s",
+			decimal10k.String(), acct.BuyingPower.String())
+	}
+
+	service := Service{tdClient: tdClient, mdClient: mdClient, acct: acct}
+	return &service, nil
+}
+
+func main() {
+	svc, err := NewService()
+	if err != nil {
+		log.Fatalf("failed to intilize service: %v", err)
+	}
+
+	underlying := "INTC"
+	td, err := svc.mdClient.GetLatestTrade(underlying, marketdata.GetLatestTradeRequest{Feed: "iex"})
```

Comment:
[minor] I suggest dropping the feed parameter. It defaults to the best feed available for the user.

Reply:
7f5ef5cad60686b69fb44fc685de6a282e972c21